File: SquaresOfASortedArray.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] nums2 = new int[nums.length];
        for(int i = 0; i < nums.length; i++){
            int square = nums[i] * nums[i];
                nums2[i] = square;
        }
        Arrays.sort(nums2);
        return nums2;
        
    }
}

/*

Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.


    Easy version without pointers
    using the function Arrays.sort() tor organized in non decresead order..

*/

File: SquaresOfASortedArray.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        std::vector<int> nums2;
        for(int numbers : nums){
            int squares = numbers * numbers;
            nums2.push_back(squares);
        }
        std::sort(nums2.begin(), nums2.end());
        return nums2;
    }
};

/*
Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.

Solved without using pointers, squares vector is sorted with the fucntion std::sort(nums2.begin(), nums2.end());

*/
